specs/templates/table-column-hld.md

```diff
@@ -0,0 +1,133 @@
+# Table Column [Template]
+
+*Refer to the [table column type philosophy docs](/packages/nimble-components/src/table/specs/table-columns-hld.md#column-type-philosophy) for column creation policies and naming conventions.*
```

Comment:
maybe we should just move this line down to the element name section

Reply:
I put it right before the element name section, as it also gives guidance on general column type creation policy.